### ISSUE TITLE
calendar: initially load on the first month that has selectable dates

### ISSUE THREAD

**Issue by slickcoders** [eBay/ebayui-core#1924]
# Bug Report

 If I were to initialize the component with
"disableBefore": "2023-07-05",
"disableAfter": "2023-07-20"

with current date is 6/20. I see that the component is showing the month of June with all dates disabled, is there a way to have to it showing July at the start?

## eBayUI Version: x.x.x
"@ebay/ebayui-core@
  version "12.1.1"

## Description

Options proposed: 

I believe the intended behavior was to prevent users from navigating to months with only disabled dates, but if June should be shown then this needs to be rethought. The way it stands, I think there are two options:

- fix the issue as you initially proposed it, and continue preventing users from navigating to June
- keep the initial focus on the current month, and always allow for navigation up to and including the current month even if it is completely disabled

Ultimately we want lets work for our users.  Let’s go with option 1, which is more intuitive.  

In summary, please load the calendar on the month with selectable dates.  User would not be able to navigate to prior months with disabled dates.  

Thanks Luke!

## Workaround
n/a

## Screenshots
![image](https://github.com/eBay/ebayui-core/assets/137234111/b0b1647e-ee04-4916-82f9-6c9ee9d1e9fd)
